storage_service/locations/models/s3.py

```diff
@@ -128,12 +128,29 @@ def browse(self, path):
         }
 
     def delete_path(self, delete_path):
-        objects = self.resource.Bucket(self._bucket_name()).objects.filter(
+        """Delete an object from an S3 bucket. We assume an object exists, if
+        it doesn't then the generator returned by the S3 library (Boto3) cannot
+        be iterated, and we raise a StorageException.
+        """
+        if delete_path.startswith(os.sep):
+            LOGGER.info(
+                "S3 path to delete {} begins with {}; removing from path prior to deletion".format(
+                    delete_path, os.sep
+                )
+            )
+            delete_path = delete_path.lstrip(os.sep, "")
+        obj = self.resource.Bucket(self._bucket_name()).objects.filter(
             Prefix=delete_path
         )
-
-        for objectSummary in objects:
-            objectSummary.delete()
+        items = False
+        for objectSummary in obj:
+            items = True
+            resp = objectSummary.delete()
+            LOGGER.info("S3 response when attempting to delete: %s", resp)
+        if not items:
+            err_str = "No packages found in S3 at: {}".format(delete_path)
+            LOGGER.warning(err_str)
```

Comment:
Aren't exceptions logged anyways? I.e. do we need to both log a warning and raise?

Reply:
It is a good point. I have been tempted to log an issue about this, it's inconsistent. Because it's the responsibility of the caller to decide what to do with an exception, the caller might not always log it back to `stderr`. In this case, if I force the error, we get a UI output, but no log output, so I create a log at the point before the instead. So we just see the following on the command line:
```
archivematica-storage-service_1  | DEBUG     2019-06-13 22:57:18  locations.models.async_manager:async_manager:_watchdog_loop:122:  Watchdog sees 0 tasks running
archivematica-storage-service_1  | INFO      2019-06-13 22:57:18  locations.models.s3:s3:delete_path:138:  S3 path to delete /will-not-be-deleted/5d93/dda0/055d/4cf9/8304/129f/3ab3/a699/rs-s3-1-5d93dda0-055d-4cf9-8304-129f3ab3a699.7z begins with /; removing from path prior to deletion
archivematica-storage-service_1  | WARNING   2019-06-13 22:57:19  locations.models.s3:s3:delete_path:152:  No packages found in S3 at: will-not-be-deleted/5d93/dda0/055d/4cf9/8304/129f/3ab3/a699/rs-s3-1-5d93dda0-055d-4cf9-8304-129f3ab3a699.7z
```
**Example UI output:**
![image](https://user-images.githubusercontent.com/1880412/59486839-03f4b080-8e7b-11e9-8336-9aae6461e88c.png)

Reply:
Sounds good. Too bad about the inconsistency, but in that case better to log as well.